src/highdicom/seg/sop.py

```diff
@@ -565,23 +573,42 @@ def __init__(
                 x_index = plane_position_names.index(
                     'XOffsetInSlideCoordinateSystem'
                 )
-                x_offset = plane_position_values[first_frame_index, x_index]
+                x_origin = plane_position_values[first_frame_index, x_index]
                 y_index = plane_position_names.index(
                     'YOffsetInSlideCoordinateSystem'
                 )
-                y_offset = plane_position_values[first_frame_index, y_index]
-                origin_item = Dataset()
-                origin_item.XOffsetInSlideCoordinateSystem = x_offset
-                origin_item.YOffsetInSlideCoordinateSystem = y_offset
-                self.TotalPixelMatrixOriginSequence = [origin_item]
-                self.TotalPixelMatrixRows = int(
-                    plane_position_values[last_frame_index, row_index] +
-                    self.Rows
-                )
-                self.TotalPixelMatrixColumns = int(
-                    plane_position_values[last_frame_index, col_index] +
-                    self.Columns
+                y_origin = plane_position_values[first_frame_index, y_index]
+                z_index = plane_position_names.index(
+                    'ZOffsetInSlideCoordinateSystem'
                 )
+                z_origin = plane_position_values[first_frame_index, z_index]
+
+                if is_tiled:
+                    origin_item = Dataset()
+                    origin_item.XOffsetInSlideCoordinateSystem = x_origin
+                    origin_item.YOffsetInSlideCoordinateSystem = y_origin
```

Comment:
These are DS attributes. Until we have a better solution, make sure to convert to string explicitly with `pydicom.valuerep.format_number_as_ds` to ensure there is no overflow.

Reply:
Good catch! Addressed via 18b2e3ef9b24aefb20271ef98f8e3bbb655fe588